src/main/java/nl/tudelft/contextproject/model/level/Room.java

```diff
@@ -0,0 +1,201 @@
+import nl.tudelft.contextproject.model.level.util.Vec2I;
+import nl.tudelft.contextproject.util.Size;
+
+/**
+ * Class that represent a room used in level creation.
+ * 
+ * This class is a wrapper for the level reader so that it can be used with the
+ * level creation class.
+ */
+public class Room {
+
+	/**
+	 * Read format from map file.
+	 * TODO: Refactor parser classes so that size only is defined at one location.
+	 */
+	private static final Pattern PATTERN = Pattern.compile("(?<width>\\d+)x(?<height>\\d+)_.*");
+
+	/**
+	 * Room size. 
+	 */
+	public Size size;
+
+	/**
+	 * Entities in the room.
+	 */
+	public Set<Entity> entities;
+
+	/**
+	 * Lights in the room.
+	 */
+	public List<Light> lights;
+
+	/**
+	 * MazeTile map.
+	 */
+	public MazeTile[][] mazeTiles;
+
+	/**
+	 * Entrance door locations.
+	 */
+	public ArrayList<Vec2I> entranceDoorsLocations;
+
+	/**
+	 * Exit door locations.
+	 */
+	public ArrayList<Vec2I> exitDoorLocations;
+
+	/**
+	 * Original folder.
+	 */
+	public String folder;
+	
+	/**
+	 * Constructor will load room from files using RoomIO.
+	 * @param folder
+	 * 		fileName of the room
+	 */
+	public Room(String folder) {
+		this.folder = folder;
+		entities = new HashSet<Entity>();
+		lights = new ArrayList<Light>();
+		try {
+			size = getSizeFromFileName(folder);
+			mazeTiles = new MazeTile[size.getWidth()][size.getHeight()];
+			RoomParser.importFile(folder, mazeTiles, entities, lights, 0, 0);	
+			setDoors();
+		} catch (IOException e) {
+			Logger.getLogger("MazeGeneration").severe("Unable to correctly read name!");
+		}
+	}
+
+	/**
+	 * Get size of room from folder name.
+	 * @param folder
+	 * 		folder of the Room
+	 * @return
+	 * 		size of room
+	 * @throws IOException 
+	 * 		if pattern of roomFolder does not match
+	 */
+	protected Size getSizeFromFileName(String folder) throws IOException {
+		int width = 0;
+		int height = 0;
+		String fileName = RoomParser.getMapFile(folder).getName();
+		Matcher m = PATTERN.matcher(fileName);
+		if (m.matches()) {
+			width = Integer.parseInt(m.group("width"));
+			height = Integer.parseInt(m.group("height"));
+			return size = new Size(width, height);
+		} else {
+			throw new IOException("Expected a .crf file present in the folder with the correct name");
+		}
+	}
+
+	/**
+	 * Set the entrance and exit doors of the room.
+	 * Needed during graph creation.
+	 */
+	public void setDoors() {
+		entranceDoorsLocations = new ArrayList<Vec2I>();
+		exitDoorLocations = new ArrayList<Vec2I>();
+
+		for (int i = 0; i < size.getWidth(); i++) {
+			for (int j = 0; j < size.getHeight(); j++) {
+				if (mazeTiles[i][j] != null) {
+					TileType mazeTile = mazeTiles[i][j].getTileType();
+					if (mazeTile == TileType.DOOR_ENTRANCE) {
+						entranceDoorsLocations.add(new Vec2I(i, j));
+					} else if (mazeTile == TileType.DOOR_EXIT) {
+						exitDoorLocations.add(new Vec2I(i, j));
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Set mazeTiles array.
```

Comment:
please add a newline after this line, so the @ lines are seperated from the method description